src/process.rs

```diff
@@ -42,6 +42,11 @@ pub(crate) trait Runtime: Sized {
     async fn wait(&mut self) -> std::io::Result<ExitStatus>;
 }
 
+pub(crate) trait RuntimeConfig: DeserializeOwned + Default + Debug + Clone {
+    fn gpu_uuid(&self) -> Option<String>;
+    fn uses_gpu(&self) -> bool;
```

Comment:
Why do you need `uses_gpu` at all?
Why couldn't you introduce `test` function on Runtime trait and implement it differently for dummy and automatic?

Reply:
I have some basic tests for `Automatic` runtime config (so I will not change it accidentally). They use `test` cmd, which is why I needed some way to disable checking GPU. I can remove them, or turn them into unit tests.